core/src/main/java/org/opensearch/remote/metadata/client/AbstractSdkClient.java

```diff
@@ -248,4 +248,14 @@ protected boolean isGlobalResource(GetDataObjectResponse response) {
             .map(y -> Objects.equals(y, globalTenantId))
             .orElse(false);
     }
+
+    /**
+     * Throw exception when the request containing cmk. It should be used in Opensearch client.
+     * @param objectRequest The candidate request.
+     */
+    protected void verifyCMK(DataObjectRequest objectRequest) {
+        if (!Objects.isNull(objectRequest.cmkRoleArn())) {
+            throw new IllegalArgumentException("cmkRoleArn should be null since current client doesn't support cmk");
+        }
```

Comment:
Why do we need to throw an exception if it's non-null?  Can't we just ignore it when it's not relevant?

The main purpose of these clients is to prevent creating forks of implementing plugins, so having different behavior for adding this field would be problematic.